engine/consensus/matching/engine.go

```diff
@@ -231,7 +231,7 @@ func (e *Engine) consumeEvents() {
 	defer close(checkSealingTicker)
 	e.unit.LaunchPeriodically(func() {
 		checkSealingTicker <- struct{}{}
-	}, 2*time.Second, 120*time.Second)
+	}, 2*time.Second, 10*time.Second)
```

Comment:
`time.Ticker` perhaps?

Reply:
Thanks Jordan, using ticker is simpler without changing the behavior.

However, since you pointed out, I noticed the behavior was changed. 

The way it works before introducing the queue is that, the checkSealing will run every 2 seconds __after__ each run. So let's say if checkSealing takes 3 seconds to finish, then after it's finished, it will for another 2 secs to start another check.

The new behavior will trigger every 2 secs no matter how long the previous check takes. 

In the case where we have a lot of unsealed blocks, the checkSealing will take longer and longer, the changed behavior will make the checkSealing run more often than before, which I'm afraid why further slow down the node.

@AlexHentschel Maybe we should discuss about this.

Reply:
Good discussion. Some thoughts 
* `LaunchPeriodically` is essentially just a slim wrapper around `Ticker` https://github.com/onflow/flow-go/blob/0b9d0e4c2c552bb7506bbe01d73dfa8226798494/engine/unit.go#L75-L83 In addition, it integrates with the `Done` functionality of the `Unit` and nicely terminates the execution if the engine is terminated. If we use the bare-bones `Ticker`, we have to re-implement this functionality.
* Note the following important detail in the [Ticker](https://golang.org/pkg/time/#Ticker) documentation
   > The ticker will adjust the time interval or drop ticks to make up for slow receivers. 
   
   I actually tested this in the go Playground: https://play.golang.org/p/07ogCO_dqTf It nicely works as expected. 

* So I don't think it will run _more often_. If sealing takes up a long time, it will just consume more of the 2 second interval leaving less room for processing blocks.